File: languages/python/01-foundations/control-flow/example/main.py
# Module focus: Choosing between branches and repeating work with predictable control flow.
# Why it matters: the example makes it possible to select branches that cover normal and boundary
# conditions before the learner tackles the exercises.
# about.

# Fixed inputs make the consequence of not handling non-positive upper bounds before entering
# loops visible and repeatable.
value = int(input("Enter an integer: "))

if value > 0:
    print("positive")
elif value < 0:
    print("negative")
else:
    print("zero")

n = int(input("Enter N: "))

factorial = 1
for i in range(1, n + 1):
    factorial *= i

# The printed result shows whether the program can write terminating loops and reason about their
# invariants.
print(f"factorial({n}) = {factorial}")
print("Numbers 1..N:")
for i in range(1, n + 1):
    print(i)

File: languages/go/01-foundations/control-flow/example/main.go
// Module focus: Choosing between branches and repeating work with predictable control flow.
// Why it matters: the example makes it possible to select branches that cover normal and boundary
// conditions before the learner tackles the exercises.

package main

import "fmt"

// Fixed inputs make the consequence of not handling non-positive upper bounds before entering
// loops visible and repeatable.
func main() {
	// These values exercise the normal path before the exercises vary the documented boundaries.
	var value int
	// The printed result shows whether the program can write terminating loops and reason about
	// their invariants.
	fmt.Print("Enter an integer: ")
	fmt.Scanln(&value)

	if value > 0 {
		fmt.Println("positive")
	} else if value < 0 {
		fmt.Println("negative")
	} else {
		fmt.Println("zero")
	}

	var n int
	fmt.Print("Enter N: ")
	fmt.Scanln(&n)

	if n < 0 {
		fmt.Println("N must be non-negative.")
		return
	}

	factorial := 1
	for i := 1; i <= n; i++ {
		factorial *= i
	}

	fmt.Printf("factorial(%d) = %d\n", n, factorial)
	fmt.Println("Numbers 1..N:")
	for i := 1; i <= n; i++ {
		fmt.Println(i)
	}
}
